irohad/ametsuchi/impl/postgres_wsv_command.hpp

```diff
@@ -70,12 +71,33 @@ namespace iroha {
       pqxx::nontransaction &transaction_;
       logger::Logger log_;
 
-      using ExecuteType = decltype(makeExecute(transaction_, log_));
+      using ExecuteType = decltype(makeExecuteResult(transaction_, log_));
       ExecuteType execute_;
 
-      // TODO: refactor to return Result when it is introduced IR-744
-      bool execute(const std::string &statement) noexcept {
-        return execute_(statement).has_value();
+      /**
+       * Transforms result which contains pqxx to WsvCommandResult,
+       * which will have error message generated by error_generator
+       * appended to error received from given result
+       * @param result which can be received by calling execute_
+       * @param error_generator function which must generate error message
+       * to be used as a return error.
+       * Function is passed instead of string to avoid overhead of string
+       * construction in successful case.
+       * @return WsvCommandResult with combined error message
+       * in case of result contains error
+       */
+      template <typename Function>
+      WsvCommandResult makeCommandResult(
+          expected::Result<pqxx::result, std::string> result,
```

Comment:
Why it isn't passed by rvalue-ref?

Reply:
It was intended to, seems like a mistake, will fix